arklex/env/agents/openai_agent.py

```diff
@@ -149,252 +157,258 @@ def _prepare_prompt(self, state: OrchestratorState, is_speech: bool = False) ->
+                    tool_output = getattr(item, "output", "")
+                    
+                    if tool_output:
+                        # Add to both trajectories for consistency
+                        tool_message = {
+                            "role": "system",
+                            "content": f"[TOOL_OUTPUT name={tool_name}] {str(tool_output)}"
+                        }
+                        
+                        # Legacy trajectory
+                        if state.function_calling_trajectory is not None:
+                            state.function_calling_trajectory.append(tool_message)
+                        
+                        # SDK trajectory
+                        if state.openai_sdk_trajectory is not None:
+                            state.openai_sdk_trajectory.append(tool_message)
+        except Exception:
+            pass
+
+    # Tool execution is managed by the Agents SDK; no manual execution path needed here.
+
+    async def generate_response(
         self, state: OrchestratorState, stream: bool = False, is_speech: bool = False
     ) -> tuple[OrchestratorState, OpenAIAgentOutput]:
-        """Unified response generation method with optional streaming."""
+        """Unified response generation method with optional streaming using Agents SDK."""
         generation_type = (
             "speech streaming"
             if is_speech and stream
             else "streaming"
             if stream
             else "standard"
         )
-        log_context.info(f"\nGenerating {generation_type} response using the agent.")
+        log_context.info(f"\nGenerating {generation_type} response using the Agents SDK.")
         input_prompt = self._prepare_prompt(state, is_speech)
         self._add_prompt_to_trajectory(state, input_prompt)
 
-        log_context.info(f"\nagent messages: {state.function_calling_trajectory}")
-        final_chain = self.llm
-        ai_message: AIMessage = final_chain.invoke(state.function_calling_trajectory)
-
-        log_context.info(f"Generated answer: {ai_message}")
-
-        # Process tool calls first
-        self._process_tool_calls(state, ai_message)
+        try:
+            cleaned_messages: list[dict[str, Any]] = []
+            sdk_traj = state.openai_sdk_trajectory or []
+            for msg in sdk_traj:
+                role = msg.get("role")
+                content = str(msg.get("content", ""))
+                # Keep non-system, keep tool outputs, keep only the current agent's prompt
+                if role != "system" or content.startswith("[TOOL_OUTPUT") or content == input_prompt:
+                    cleaned_messages.append(msg)
+            if len(cleaned_messages) != len(sdk_traj):
+                log_context.info("Pruned prior system prompts from other agents to keep the conversation natural.")
+                state.openai_sdk_trajectory = cleaned_messages
+        except Exception:
+            pass
```

Comment:
should we log something here instead of eating the exception.
Could you clarify a bit why we need to clean the messages?